### ISSUE TITLE
Error - Unauthorized action or expired link after login

### ISSUE THREAD

**Issue by gehwissenlos** [YOURLS/YOURLS#3087]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps) guide.

### Version

1.8.2

### Description

I updated to version 1.8.2 and now a get a message "Error - Unauthorized action or expired link" every time I log in. After I hit "login" it prints this error. In the log of my webserver I see a line "Error | 403 | POST /admin/ HTTP/1.0". When I enter domain.tld/admin I'm in the backend. So login is successful but the post method throws an error. I activated debug mode but there wasn't any hint what could be the problem.
This happens on two installations of Yourls. I use PHP 8.0.11 and MySQL 8.0.26, tested with Chrome and Edge on Windows 10 and 11.

### Expectation

Successful login without error message

**Comment by gehwissenlos** [comment]
You're absolutely right. I thought it must be a problem with the core functions because it's there since the update of the yourls core files. Now I deactivated all plugins I have on both installations and found out, that the problem is linked to the sleeky backend plugin. I updated it to version 2.5.0 but still the same error message appears.
I will contact the author of this plugin, perhaps he has any ideas what causes this.
Thanks for your help.

**Comment by issteve** [comment]
I had exactly the same problem. Thankfully I found this thread here and did try it with the plugins in my installation. 
After deactivating:
"Don't Track Admins" / "Don't Track Admin Clicks" (1.2) by dgw the problem was solved. 
Just wanted to leave a note here for others finding it by an internet search like I did...

**Comment by dgw** [comment]
@issteve The error message comes from authentication handling—specifically the nonce check, which predates the existence of my plugin (but used to live in a different core file; see 970936e5744c177deba69374a0669a4c7a8cded1).

My plugin doesn't touch any of that. It's supremely simple code, really, and I suspect you've mistaken a temporal correlation for causation. But if there really is a bug in my plugin after all this time, a patch would be welcome.

**Comment by issteve** [comment]
@dgw I did check it multiple times back and forth with the same result. 
Just changing "yourls_is_valid_user" into "yourls_is_admin" did stop the error too (but it doesn't do any longer what it should and did before ;-) ). I have no idea what else could be used somehow instead of "yourls_is_valid_user".
Anyhow I didn't say there is any bug in your plugin, but deactivating helps and it still does. 
So I guess something in Yourls changed around October 2021 regarding "yourls_is_valid_user" which is also in the plugin gehwissenlos had a problem with, but I'm not into php to really judge it.

**Comment by gehwissenlos** [comment]
I think there was a change in yourls_is_valid_user because in the issue I opened in this plugin somebody wrote that the deleted the line "if (yourls_is_valid_user() != 1)" and it works. So perhaps there was a bigger change under the hood and that's the reason why several plugins have this error.

**Comment by dgw** [comment]
@issteve Sounds good, thanks for testing it.

Would be worth trying to replace `if( yourls_is_valid_user() === true ) {` (in my plugin) with `if( defined( 'YOURLS_USER' ) {`. That's set by auth validation functions and checking it doesn't require calling the validation again. YOURLS itself uses that constant to decide whether it should show a logout link.

https://github.com/YOURLS/YOURLS/blob/master/includes/functions-html.php#L752-760

**Comment by issteve** [comment]
I did check your suggestion after I figured out that there was one bracket missing. So I tried:
`if( defined( 'YOURLS_USER' ) ) {`
... and it didn't work that well. So, yes, I was able to activate it and the login-error was gone but it did count some clicks anyway. Yes, "some" so not everyone, but with all my tries at least the first one after the login. But some others as well, but not all...

**Comment by dgw** [comment]
Only "some" being logged might be due to caching of 301 redirects, rather than anything the plugin is (not) doing.

Upon further review, I don't think YOURLS defines that constant if it isn't already checking auth itself for some other reason—a request for something that requires auth, for example. Near as I can tell that never happens during a shortlink redirection.

So I'm back to square one on eliminating the issue from my plugin. If I can't call `yourls_is_valid_user()` safely, I don't know how else to check if the request came from someone who's logged in. :/

**Comment by ozh** [comment]
@dgw I'll have a look at the plugin & this issue next week-end. I don't see why a plugin would not be able to check `yourls_is_valid_user()`

**Comment by daffydock** [comment]
> You're absolutely right. I thought it must be a problem with the core functions because it's there since the update of the yourls core files. Now I deactivated all plugins I have on both installations and found out, that the problem is linked to the sleeky backend plugin. I updated it to version 2.5.0 but still the same error message appears. I will contact the author of this plugin, perhaps he has any ideas what causes this. Thanks for your help.

Just as FYI, for anyone who ends up here looking for a fix Re: the Sleeky Back end, the issue has not been addressed but someone over there came up with a fix that I will post below. Both as a way to save people time but also to elucidate the likely issues with and why some extensions may have with Yourls. 

Cheers. 

>This seems to have worked for me...
Near the bottom of user/plugins/sleekly-backend/plugin.php, change the line:
if (yourls_is_valid_user() != 1) {
to
if( defined( 'YOURLS_USER' ) ) {

Link to original post:
https://github.com/Flynntes/Sleeky/issues/127#issuecomment-1586014710
